app/assets/javascripts/auto_title.js

```diff
@@ -7,7 +7,7 @@ function path(name) {
 }
 
 function retrieveTitleSuggestion() {
-  var link = this.value;
+  var link = encodeURI(this.value);
```

Comment:
@bion adding `encodeURI()` is my attempt at protecting against XSS in the form input.  

You also mentioned being careful with what gets returned when the `<title>` element is parsed out, but as I recall when we last talked about XSS, Rails automatically escapes any dangerous characters with the `sanitize` method.